api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -492,6 +523,14 @@ public String doc() {
       return doc;
     }
 
+    public JsonNode initialDefaultValue() {
```

Comment:
This API should not use Jackson classes. Instead, it should use the standard Iceberg in-memory representation:
* boolean - Java primitive boolean
* int - Java primitive int
* long - Java primitive long
* float - Java primitive float
* double - Java primitive double
* date - Java primitive int
* time - Java primitive long
* timestamp/timestamptz - Java primitive long
* string - Java CharSequence
* uuid - Java UUID
* binary - Java ByteBuffer
* fixed(L) - Java ByteBuffer
* decimal(P, S) - Java BigDecimal
* list - Java Collection
* struct - StructLike
* map - Java Map

Reply:
> * list - Java Collection

I think using Java `List` is more correct, as List cares about the order.



> * struct - StructLike

I see this comment:

```
/**
 * Interface for accessing data by position in a schema.
 * <p>
 * This interface supports accessing data in top-level fields, not in nested fields.
 */
```
in that interface, also I see the API exposes is getting fields by position, which doesn't suit our casing of accessing fields by field id.

I think using a native Java map would be better.